server/src/test/java/org/elasticsearch/common/util/set/SetsTests.java

```diff
@@ -94,6 +95,11 @@ public void testCapacityIsEnoughForSetToNotBeResized() {
         }
     }
 
+    public void testAddToCopy() {
+        assertThat(addToCopy(Set.of("a", "b"), "c"), containsInAnyOrder("a", "b", "c"));
+        assertThat(addToCopy(Set.of("a", "b"), "c", "d"), containsInAnyOrder("a", "b", "c", "d"));
+    }
+
```

Comment:
I guess it would be nice to test that we don't break some fundamental properties for the resulted set:
```java
        String[] subset = randomSubsetOf(set).toArray(String[]::new);
        assertEquals(addToCopy(set, subset), Sets.union(set, Set.copyOf(Arrays.asList(subset))));
        assertEquals(set, addToCopy(set, set.toArray(new String[] {})));
        assertEquals(set, addToCopy(set));
```